ouroboros-consensus/test/storage-test/Test/Ouroboros/Storage/PerasCertDB/StateMachine.hs

```diff
@@ -113,7 +113,15 @@ instance StateModel Model where
 
   precondition (Model model) = \case
     OpenDB -> not model.open
-    _ -> model.open
+    action ->
+      model.open && case action of
+        CloseDB -> True
+        -- Do not add equivocating certificates.
```

Comment:
Don't we actually want to test the addition of equivocated certificates?

Reply:
I think he was relying on (Section 2.2.2 of the [design document](https://tweag.github.io/cardano-peras/peras-design.pdf):

> Assumption
> For every Peras round, there is at most one block that has a valid certificate in that round.
>
> This assumption is justified by an appropriate parameterization of perasQuorum and perasN, such that an adversary never attains sufficient weight in the committee to equivocate certificates.

Under that assumption, I think it makes sense to establish such a precondition. We probably want to justify it better in the code, though. E.g.:

> Equivocating certificates are not possible by construction.
> See https://tweag.github.io/cardano-peras/peras-design.pdf (section 2.2.2)